pkg/util/iptables/testing/fake.go

```diff
@@ -26,80 +26,106 @@ import (
 )
 
 const (
+	// Destination represents the destination address flag
 	Destination = "-d "
-	Source      = "-s "
-	DPort       = "--dport "
-	Protocol    = "-p "
-	Jump        = "-j "
-	Reject      = "REJECT"
-	ToDest      = "--to-destination "
-	Recent      = "recent "
-	MatchSet    = "--match-set "
-	SrcType     = "--src-type "
-	Masquerade  = "MASQUERADE "
+	// Source represents the source address flag
+	Source = "-s "
+	// DPort represents the destination port
+	DPort = "--dport "
+	// Protocol represents the protocol flag which takes input by number of name
+	Protocol = "-p "
+	// Jump represents jump flag specifies the jump target
+	Jump = "-j "
+	// Reject specifies the reject target
+	Reject = "REJECT"
+	// ToDest represents the --to-destination flag used to specify the destination address in DNAT
+	ToDest = "--to-destination "
+	// Recent represents the sub-command recent that allows to dynamically create list of IP address to match against
+	Recent = "recent "
+	// MatchSet represents the --match-set flag which match packets against the specified set
+	MatchSet = "--match-set "
+	// SrcType represents the --src-type flag which matches if the source address is of given type
+	SrcType = "--src-type "
+	// Masquerade represents the target that is used in nat table.
+	Masquerade = "MASQUERADE "
 )
 
+// Rule holds a map of rules.
 type Rule map[string]string
 
-// no-op implementation of iptables Interface
+// FakeIPTables no-op implementation of iptables Interface.
 type FakeIPTables struct {
 	hasRandomFully bool
 	Lines          []byte
 }
 
+// NewFake returns a pointer for no-op implementation of iptables Interface.
 func NewFake() *FakeIPTables {
 	return &FakeIPTables{}
 }
 
+// SetHasRandomFully will enable the port maping fully randomized in the no-op implementation of iptables Interface.
 func (f *FakeIPTables) SetHasRandomFully(can bool) *FakeIPTables {
```

Comment:
the convention here is to just say "`SetHasRandomFully is part of iptables.Interface`" because the interface method is already documented in the interface definition, and if we try to document it here as well it's just going to get out of sync

Reply:
I understood now. I did a little overkill on the comments and your suggestion makes perfect sense. As this is a testing package the comments/doc should be clear on the actual package.

I changed it for all the interface methods.